File: projects/06/java/PrintArray.java
/* PrintArray.java is a driver for function PrintArray().
 * Author: Sarah Whitten
 * Date: March 11, 2019
 ****************************************************************/

import java.util.*;

public class PrintArray {

	public static void main(String[] args) {
		Scanner s = new Scanner(System.in);
    	System.out.print("Enter the size of your array:\n");

    	int size = s.nextInt();
    	double ourArray[];
    	ourArray = new double [size];
    	ReadArray(ourArray, size, s);
    	PrintArray(ourArray, size);
	}


	public static void PrintArray(double[] anArray, int size) {
		System.out.println("Here are the contents of your array: ");
		for (int i = 0; i < size; i++) {
			System.out.println(anArray[i]);
		}
	}

	public static void ReadArray(double[] anArray, int size, Scanner scan) {
		System.out.println("Enter the contents of your array: ");
		for (int i = 0; i < size; i++) {
			anArray[i] = scan.nextDouble();
		}
	}
}

File: projects/06/clojure/src/printArray.clj
;; print.clj "test-drives" function printArray().
;;
;; Output: the contents of the user-created array.
;;
;; Usage: clojure -m print
;;
;; Created by: Sarah Whitten
;; Date: March 11, 2019
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns printArray)

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;; printArray() displays the array values.       
;; Receive: anArray, a vector; size, an integer.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

;; I tested this separately from the readArray() function
;; and it works fine. I just can't figure out how to modify
;; the array and spit it back out to be printed
(defn printArray [anArray size]
	(println "Here are the contents of your array:")
	(loop [i 0]
		(when (< i size)
			(println (get anArray i))
			(recur (+ i 1))
		)
	)
)

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;; readArray() assigns values to the array     
;; Receive: size, an integer
;; Return: anArray, a vector containing the correct values
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

;; WHY WON'T IT WORK
(defn readArray [size]
	(println "Enter the contents of your array:")
	(let
		[ anArray (vector) ]
		(loop [i 0]
			(when (< i size)
				(concat anArray (double (read)))
				(recur (+ i 1))
			)
		)
		anArray
	)
)

(defn -main []
	(println "Enter the size of your array:")
	(let
		[ size (int (read))
		  ourArray (readArray size)
		  ; testArray [0.0, 1.0]
		]
		; (println ourArray)
		; (println testArray)
		; (printArray testArray size)
		(printArray ourArray size)
	)
)
